app/test/vesting_test.go

```diff
@@ -0,0 +1,632 @@
+	gAccounts := authtypes.GenesisAccounts{}
+	for i := range vAccounts {
+		gAccounts = append(gAccounts, authtypes.GenesisAccount(vAccounts[i]))
+	}
+
+	return func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		gs, err := AddGenesisAccountsWithBalancesToGenesisState(gs, gAccounts, balances)
+		assert.NoError(s.T(), err)
+		return gs
+	}
+}
+
+// initContinuousVestingAccounts function initializes continuous vesting accounts for testing purposes.
+// It takes the count of accounts as input and returns a GenesisOption. It generates account names,
+// base accounts, and balances. It defines start & end times to creates vesting accounts. The function
+// then updates the genesis state with the generated vesting accounts and balances. The start & endTime
+// of each account increases progressively to ensure some accounts have locked balances, catering to the
+// testing requirements.
+func (s *VestingModuleTestSuite) initContinuousVestingAccounts(count int) testnode.GenesisOption {
+	initCoinsForGasFee := sdk.NewCoin(app.BondDenom, sdk.NewInt(initBalanceForGasFee))
+	names := testfactory.GenerateAccounts(count)
+	s.accounts.Store(ContinuousVestingAccountType, accountDispenser{names: names})
+
+	vAccounts, balances := testfactory.GenerateContinuousVestingAccounts(s.kr, names, initCoinsForGasFee)
+	gAccounts := authtypes.GenesisAccounts{}
+	for i := range vAccounts {
+		gAccounts = append(gAccounts, authtypes.GenesisAccount(vAccounts[i]))
+	}
+
+	return func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		gs, err := AddGenesisAccountsWithBalancesToGenesisState(gs, gAccounts, balances)
+		assert.NoError(s.T(), err)
+		return gs
+	}
+}
+
+// unusedAccount returns an unused account name of the specified account type
+// unusedAccount returns an unused account name of the specified account type
+// for the VestingModuleTestSuite. If the account type is not found, it returns
+// an error.
+func (s *VestingModuleTestSuite) unusedAccount(accType accountType) (string, error) {
+	s.accountsMut.Lock()
+	defer s.accountsMut.Unlock()
+
+	accountsAny, found := s.accounts.Load(accType)
+	if !found {
+		return "", fmt.Errorf("account type `%s` not found", accType.String())
+	}
+
+	accounts := accountsAny.(accountDispenser)
+	if accounts.counter >= len(accounts.names) {
+		return "", fmt.Errorf("out of unused accounts for type `%s`", accType.String())
+	}
+
+	name := accounts.names[accounts.counter]
+	accounts.counter++
+	s.accounts.Store(accType, accounts)
+
+	return name, nil
+}
+
+// getAnUnusedContinuousVestingAccount returns an unused continuous vesting account and its name.
+//
+// It takes a minimum start-time as input and finds an unused account whose start time is greater than the input.
+func (s *VestingModuleTestSuite) getAnUnusedContinuousVestingAccount(minStartTime int64) (vAcc vestingtypes.ContinuousVestingAccount, name string, err error) {
```

Comment:
[optional][refactor]
Same here, we could refactor these 3 functions with one function and a cast after returning the value:
```
func (s *VestingModuleTestSuite) getNonRegularUnusedVestingAccount(accType accountType, minStartTime int64) (vAcc authtypes.AccountI, name string, err error) {
	for {
		switch accType {
		case RegularAccountType:
			return error
		case ContinuousVestingAccountType:
			...
		}
```

And then, casting like:
```
	acc, name, err := s.getNonRegularUnusedVestingAccount(ContinuousVestingAccountType, tmtime.Now().Unix() - 5)
	assert.NoError(s.T(), err)
	vAcc := acc.(*vestingtypes.ContinuousVestingAccount)
```

Reply:
We need to pass `minEndTime` to the delayed vesting account function, so if we merge those, it might introduce confusion.